splunk-otel-android/src/main/java/com/splunk/rum/DeviceSpanStorageLimiter.java

```diff
@@ -0,0 +1,92 @@
+package com.splunk.rum;
+
+import static com.splunk.rum.SplunkRum.LOG_TAG;
+import static java.util.Comparator.comparingLong;
+
+import android.util.Log;
+
+import java.io.File;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class DeviceSpanStorageLimiter {
+    static final int DEFAULT_MAX_STORAGE_USE_MB = 25;
+    private final File path;
+    private final int maxStorageUseMb;
+    private final FileUtils fileUtils;
+
+    private DeviceSpanStorageLimiter(Builder builder) {
+        this.path = builder.path;
+        this.maxStorageUseMb = builder.maxStorageUseMb;
+        this.fileUtils = builder.fileUtils;
+    }
+
+    /**
+     * Ensures that the storage currently used by spans has not exceeded the limit.
+     * If it does, it will delete older files until the limit is no longer exceeded.
+     *
+     * @return - true if the free space is under the limit (including when files have
+     * been deleted to return back under the limit), false if not enough space could be
+     * freed to get us back under out limit.
+     */
+    boolean ensureFreeSpace() {
+        tryFreeingSpace();
+        return path.getFreeSpace() > limitInBytes();
```

Comment:
I don't really get this condition -- it's comparing `the number of unallocated bytes in the partition` (free disk space if I understand it correctly) against our own usage limit (25 MB). 
What if for some reason we did not manage to remove old span files and free space in our "spans dir", but there's still 1gb more on the disk? Should we try to write the spans in this case?
Or, the disk partition is really small, like 20MB, and we've managed to free all space? Should we not write the spans, if we're under the limit?

Reply:
I knew this was going to come up..... 🙃 

The idea was to do two things:
1. Delete old telemetry if we're using more than a certain amount on disk.
2. Play nice if the disk is getting full

So with #2, we are simply checking to see if the disk is getting full -- and we do this by seeing if the amount of free space is less than our max size limit. 

The downside of this is that the limit is now doing double duty -- but I definitely didn't think another config item was warranted and I didn't want to hard code something. Maybe it's confusing....I couldn't think of a great way to make it clearer (open to ideas here).

Reply:
I think it's fine if we just add a comment explaining the reason for this, and that it's intentional.